lib/galaxy/datatypes/genetics.py

```diff
@@ -827,6 +829,637 @@ def __init__(self, **kwd):
                                 substitute_name_with_metadata='base_name', is_binary=True)
 
 
+class LinkageStudies(Text):
+    """
+    superclass for classical linkage analysis suites
+    """
+    test_files = [
+        'linkstudies.allegro_fparam', 'linkstudies.allegro_ihaplo',
+        'linkstudies.alohomora_gts', 'linkstudies.alohomora_maf',
+        'linkstudies.alohomora_map', 'linkstudies.alohomora_ped',
+        'linkstudies.linkage_datain', 'linkstudies.linkage_map',
+        'linkstudies.linkage_pedin'
+    ]
+
+    def __init__(self, **kwd):
+        Text.__init__(self, **kwd)
+        self.lcount = 0
+        self.max_lines = 10
+        # iterate whole file without errors
+        self.eof_res = True
+
+    @staticmethod
+    def __is_binary_file(fstream):
+        """
+        Private binary file tester
+        """
+        result = False
+        if '\x00' in fstream.readline(512):
+            result = True
+
+        fstream.seek(0)
+        return result
+
+    @staticmethod
+    def tokenizer(line, sep=None):
+        """
+        General purpose string tokenizer
+        """
+        if sep is None:
+            return line.splitlines()[0].split(sep)
+
+        return line.splitlines()[0].split(sep)
+
+    def eof_function(self):
+        """
+        Overridable end-of-file function
+        """
+        return self.eof_res
+
+    def __per_line_op(self, line):
+        """
+        Private per-line operation and line counter
+        """
+        self.lcount += 1
+        if self.lcount > self.max_lines:
+            return -1
+
+        return self.line_op(line)
+
+    def line_op(self, line):
+        """
+        Overridable per line operation
+        """
+        return None
+
+    def header_check(self, fio):
+        """
+        Overrideable post-binary file check function
+        """
+        return True
+
+    def sniffer(self, filename):
+        with open(filename, "r") as fio:
+
+            if LinkageStudies.__is_binary_file(fio):
+                return False
+
+            if not self.header_check(fio):
+                return False
+
+            for line in fio:
+                line_res = self.__per_line_op(line)
+
+                if line_res == -1:
+                    # run eof_function
+                    break
+
+                if line_res is not None:
+                    return line_res
+
+            return self.eof_function()
+        return False
+
+
+class PreMakePed(LinkageStudies):
+    """
```

Comment:
can you have a look at https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Fformat_1915

And see if your filetypes have a EDAM ontology term?